virtcontainers/filesystem_resource_storage.go

```diff
@@ -1,873 +0,0 @@
-func (fs *filesystem) createSandboxTempFile(sandboxID string) (string, error) {
-	if sandboxID == "" {
-		return "", errNeedSandboxID
-	}
-
-	dirPath := filepath.Join(runStoragePath, sandboxID, "tempDir")
-	err := os.MkdirAll(dirPath, dirMode)
-	if err != nil {
-		return "", err
-	}
-
-	f, err := ioutil.TempFile(dirPath, "tmp-")
-	if err != nil {
-		return "", err
-	}
-
-	return f.Name(), nil
-}
-
-// storeDeviceIDFile is used to marshal and store device IDs to disk.
-func (fs *filesystem) storeDeviceIDFile(file string, data interface{}) error {
-	if file == "" {
-		return errNeedFile
-	}
-
-	f, err := os.Create(file)
-	if err != nil {
-		return err
-	}
-	defer f.Close()
-
-	devices, ok := data.([]ContainerDevice)
-	if !ok {
-		return fmt.Errorf("Incorrect data type received, Expected []string")
-	}
-
-	jsonOut, err := json.Marshal(devices)
-	if err != nil {
-		return fmt.Errorf("Could not marshal devices: %s", err)
-	}
-
-	if _, err := f.Write(jsonOut); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// storeDeviceFile is used to provide custom marshalling for Device objects.
-// Device is first marshalled into TypedDevice to include the type
-// of the Device object.
-func (fs *filesystem) storeDeviceFile(file string, data interface{}) error {
-	if file == "" {
-		return errNeedFile
-	}
-
-	f, err := os.Create(file)
-	if err != nil {
-		return err
-	}
-	defer f.Close()
-
-	devices, ok := data.([]api.Device)
-	if !ok {
-		return fmt.Errorf("Incorrect data type received, Expected []Device")
-	}
-
-	var typedDevices []TypedDevice
-	for _, d := range devices {
-		tempJSON, _ := json.Marshal(d)
-		typedDevice := TypedDevice{
-			Type: string(d.DeviceType()),
-			Data: tempJSON,
-		}
-		typedDevices = append(typedDevices, typedDevice)
-	}
-
-	jsonOut, err := json.Marshal(typedDevices)
-	if err != nil {
-		return fmt.Errorf("Could not marshal devices: %s", err)
-	}
-
-	if _, err := f.Write(jsonOut); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func (fs *filesystem) fetchFile(file string, resource sandboxResource, data interface{}) error {
-	if file == "" {
-		return errNeedFile
-	}
-
-	fileData, err := ioutil.ReadFile(file)
-	if err != nil {
-		return err
-	}
-
-	switch resource {
-	case devicesFileType:
-		devices, ok := data.(*[]api.Device)
-		if !ok {
-			return fmt.Errorf("Could not cast %v into *[]Device type", data)
-		}
-
-		return fs.fetchDeviceFile(fileData, devices)
-	}
-
-	return json.Unmarshal(fileData, data)
-}
-
-// fetchDeviceFile is used for custom unmarshalling of device interface objects.
-func (fs *filesystem) fetchDeviceFile(fileData []byte, devices *[]api.Device) error {
-	var typedDevices []TypedDevice
-	if err := json.Unmarshal(fileData, &typedDevices); err != nil {
-		return err
-	}
-
-	var tempDevices []api.Device
-	for _, d := range typedDevices {
-		l := fs.Logger().WithField("device-type", d.Type)
-		l.Info("Device type found")
-
-		switch d.Type {
-		case string(config.DeviceVFIO):
```

Comment:
Where have you moved specific fetch implementation related to devices? I could not find it in `virtcontainers/store/filesystem.go`

Reply:
Right, I'm fixing that now...

Reply:
Fixed now.

Reply:
I need to add a lot more unit tests to `vc_test.go`